clients/src/main/java/org/apache/kafka/clients/consumer/internals/MembershipManagerImpl.java

```diff
@@ -587,6 +617,14 @@ public void transitionToFatal() {
             return;
         }
 
+        if (previousState == MemberState.LEAVING || previousState == MemberState.PREPARE_LEAVING) {
+            log.info("Member {} with epoch {} was leaving the group with state {} when it got a " +
+                "fatal error from the broker. It will discard the ongoing leave and remain in {} " +
+                "state.", memberId, memberEpoch, previousState, state);
+            maybeCompleteLeaveInProgress();
```

Comment:
Per the above comment, there's no immediate checks around this call to `maybeCompleteLeaveInProgress()` that the state is `UNSUBSCRIBED`. From my reading of it, line 606 sets the state to `FATAL`.

Again, not trying to nitpick here, I promise 😄 I'm just trying to understand the code and the expected state(s) when calling `maybeCompleteLeaveInProgress()`.

Reply:
All in the above response. The state is indeed not `UNSUBSCRIBE` here, the state is FATAL and we do have to `maybeCompleteLeaveInProgress` (to make sure we complete a leave operation that was running before we got the fatal error in a HB response while let's say, running onPartitionsRevoked callback to leave the group). Makes sense?

Reply:
nit: I think `state` will always be `MemberState.FATAL` here, so I would use it directly (as in the log messages above)